hpx/parallel/algorithms/fill.hpp

```diff
@@ -65,10 +62,10 @@ namespace hpx { namespace parallel { inline namespace v1
               : fill::algorithm("fill")
             {}
 
-            template <typename ExPolicy, typename InIter, typename T>
+            template <typename ExPolicy, typename FwdIter, typename T>
             HPX_HOST_DEVICE
             static hpx::util::unused_type
-            sequential(ExPolicy, InIter first, InIter last,
+            sequential(ExPolicy, FwdIter first, FwdIter last,
```

Comment:
Yes, this could be left as `InIter`, but it will never be called with an input iterator... Not sure how to proceed.